# EDA
 Exploring Data

In [1]:
import pandas as pd

In [3]:
all_players = pd.read_csv('../data/all_players.csv')
all_goalkeepers = pd.read_csv('../data/all_goalkeepers.csv')

## Players dataframe
This dataframe excludes goalkeepers

In [14]:
print('min year: ', all_players.Year.min(), ' ', 'max year: ', all_players.Year.max(),)
print(all_players.shape)
print(all_players.info())
all_players.head()

min year:  1996   max year:  2020
(15767, 28)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15767 entries, 0 to 15766
Data columns (total 28 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Player   15767 non-null  object 
 1   Club     10274 non-null  object 
 2   POS      15767 non-null  object 
 3   GP       15767 non-null  int64  
 4   GS       15767 non-null  int64  
 5   MINS     15767 non-null  int64  
 6   G        15767 non-null  int64  
 7   A        15767 non-null  int64  
 8   SHTS     15767 non-null  int64  
 9   SOG      15767 non-null  int64  
 10  GWG      15767 non-null  int64  
 11  PKG/A    15767 non-null  object 
 12  HmG      15767 non-null  int64  
 13  RdG      15767 non-null  int64  
 14  G/90min  15767 non-null  float64
 15  SC%      15767 non-null  float64
 16  GWA      15767 non-null  int64  
 17  HmA      15767 non-null  int64  
 18  RdA      15767 non-null  int64  
 19  A/90min  15767 non-null  float64
 20  FC  

Player Club POS  GP  GS  MINS   G   A  SHTS  SOG  ...  RdA  \
0     Roy Lassiter  TB    F  30  30  2580  27   4    76   49  ...    2   
1   Raul Diaz Arce  DC    F  28  28  2351  23   2   100   49  ...    2   
2  Eduardo Hurtado  LA    F  26  26  2323  21   7    87   56  ...    3   
3            Preki  KC    M  32  32  2880  18  13   140   61  ...    4   
4    Brian McBride  CLB   F  28  28  2307  17   3    79   44  ...    1   

  A/90min  FC  FS  OFF  YC  RC   SOG%  Year  Season  
0    0.14  20  39   70   2   0  64.47  1996     reg  
1    0.08  32  26   35   6   1  49.00  1996     reg  
2    0.27  48  26   25   5   0  64.37  1996     reg  
3    0.41  26  44    7   3   0  43.57  1996     reg  
4    0.12  21  46   10   0   0  55.70  1996     reg  

[5 rows x 28 columns]

#### Important Columns:
Player <br>
Club  <br>
POS - position <br>
G - goals <br>
A - assists <br>
G/90 min - goals per game <br>
A/90 min - assists per game <br>
SOG% - percentage of shots on goal <br>
Year <br>
Season - regular or post

#### Data Frame info:
Total Rows: 15767
<br><br>
Total Columns: 28
<br><br>
Range of Years: 1996 - 2020
<br><br>
Club has 5493 nulls
<br><br>
SOG% has 5070 nulls

In [15]:
all_players.POS.unique()

array(['F', 'M', 'D', 'M-F', 'D-M', 'M-D', 'F-M'], dtype=object)

## Goal Keepers dataframe

In [17]:
print('min year: ', all_goalkeepers.Year.min(), ' ', 'max year: ', all_goalkeepers.Year.max())
print(all_goalkeepers.shape)
print(all_goalkeepers.info())
all_goalkeepers.head()

min year:  1996   max year:  2020
(2006, 19)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2006 entries, 0 to 2005
Data columns (total 19 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  2006 non-null   object 
 1   Club    1324 non-null   object 
 2   POS     2006 non-null   object 
 3   GP      2006 non-null   int64  
 4   GS      2006 non-null   int64  
 5   MINS    2006 non-null   int64  
 6   SHTS    2006 non-null   int64  
 7   SV      2006 non-null   int64  
 8   GA      2006 non-null   int64  
 9   GAA     2006 non-null   float64
 10  PKG/A   2006 non-null   object 
 11  W       2006 non-null   int64  
 12  L       2006 non-null   int64  
 13  T       2006 non-null   int64  
 14  ShO     2006 non-null   int64  
 15  W%      2006 non-null   float64
 16  Sv%     2006 non-null   float64
 17  Year    2006 non-null   int64  
 18  Season  2006 non-null   object 
dtypes: float64(3), int64(11), object(5)
memory usage: 297.9+ KB
None


Player Club POS  GP  GS  MINS  SHTS   SV  GA   GAA PKG/A   W   L  \
0       Mark Dodd  DAL  GK  31  31  2776   235  160  45  1.45   5/5  17  14   
1      Tony Meola  MET  GK  29  29  2610   188  142  38  1.31   3/4  14  15   
2  Mark Dougherty  TB   GK  28  28  2520   186  123  47  1.68  9/10  17  11   
3    Jorge Campos  LA   GK  24  24  2160   133  100  27  1.13   0/1  13   8   
4  Garth Lagerwey  KC   GK  23  22  1959   133   83  38  1.73   2/2  12  10   

   T  ShO    W%   Sv%  Year Season  
0  0    6  54.8  68.1  1996    reg  
1  0    9  48.3  75.5  1996    reg  
2  0    3  60.7  66.1  1996    reg  
3  0    4  54.2  75.2  1996    reg  
4  0    3  52.2  62.4  1996    reg

#### Important Columns:
Player <br>
Club  <br>
POS - position <br>
GA - goals scored against keeper <br>
SV - goals saved <br>
PKG/A - penatly kick saves ratio <br>
Sv% - percentage of goals saved <br>
Year <br>
Season - regular or post

#### Data Frame info:
Total Rows: 2006
<br><br>
Total Columns: 19
<br><br>
Range of Years: 1996 - 2020
<br><br>
Club has 682 nulls


In [16]:
all_goalkeepers.POS.unique()

array(['GK'], dtype=object)